### PR TITLE
Clean up CARGO_CFG_TARGET_FAMILY variable parsing 

### PR DESCRIPTION
**Pull request by ravenexp**
Improves error reporting in e.g. https://github.com/PyO3/pyo3/issues/1522

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -518,13 +516,15 @@ fn windows_hardcoded_cross_compile(
 fn load_cross_compile_info(
     cross_compile_config: CrossCompileConfig,
 ) -> Result<(InterpreterConfig, BuildFlags)> {
-    let target_family = env::var("CARGO_CFG_TARGET_FAMILY")?;
-    // Because compiling for windows on linux still includes the unix target family
-    if target_family == "unix" {
+    match env::var("CARGO_CFG_TARGET_FAMILY") {
```
Comment (davidhewitt) [inline build.rs:519]: Hmm the `Err` can include utf8 decoding errors too. We can avoid those using `var_os`:

```suggestion
    match env::var_os("CARGO_CFG_TARGET_FAMILY") {
```

File: build.rs [hunk 2]
```diff
@@ -518,13 +516,15 @@ fn windows_hardcoded_cross_compile(
...
-        load_cross_compile_from_sysconfigdata(cross_compile_config)
-    } else {
-        windows_hardcoded_cross_compile(cross_compile_config)
+        Ok(os) if os == "unix" => load_cross_compile_from_sysconfigdata(cross_compile_config),
+        // Use hardcoded interpreter config when targeting Windows
+        Ok(os) if os == "windows" => windows_hardcoded_cross_compile(cross_compile_config),
+        // ???
+        Ok(os) => Err(format!("Unsupported target OS family: {}", os).into()),
+        // wasm32-wasi / bare metal ?
+        Err(_) => Err("CARGO_CFG_TARGET_FAMILY is not set when cross-compiling".into()),
```
Comment (davidhewitt) [inline build.rs:527]: With `var_os`:

```suggestion
        Some(os) if os == "unix" | None => load_cross_compile_from_sysconfigdata(cross_compile_config),
        // Use hardcoded interpreter config when targeting Windows
        Some(os) if os == "windows" => windows_hardcoded_cross_compile(cross_compile_config),
        // ???
        Some(os) => bail!("Unsupported target OS family for cross-compilation: {:?}", os),
```

I also moved the "missing" case to just load sysconfigdata - I don't see any harm in trying to load that, and it's probably better than just giving up.

**Review by davidhewitt — COMMENTED** [review]
Thanks for the tidy up!

Couple of suggestions on the `match` statement...

**Comment by ravenexp** [comment]
Oh, I expected everything coming from Cargo to be UTF-8-safe, but sure, it looks cleaner this way.

**Review by davidhewitt — APPROVED** [review]
This looks great now. Thanks very much for the PR!

**Comment by davidhewitt** [comment]
> I expected everything coming from Cargo to be UTF-8-safe

I would think Cargo is, but I have no idea if a user can ever override these variables with non-utf8 content 🤷

**Pull request merged**